src/legacy/ui/public/state_management/app_state.d.ts

```diff
@@ -18,4 +18,9 @@
  */
 import { State } from './state';
 
-export type AppState = State;
+export class AppState extends State {}
+
+export type IAppState<
```

Comment:
We pass around the class AppState as well as instances of the class AppState.  This is confusing, but IAppState I made for the class, AppState for the instance of the class. Open to better suggestions.

Reply:
I'll do `TAppStateClass`.  The problem is that I have to use `AppStateClass`, in dashboard_state_manager as a variable passed around but I can't lowercase the a because I do `new AppStateClass`.